src/site/pages/digital/src/containers/SelectionPopup/modules/BusButtonModule.tsx

```diff
@@ -13,23 +16,83 @@ type Props = {
 export const BusButtonModule = ({ info }: Props) => {
     const { renderer, history } = info;
 
-    const [props, ps] = useSelectionProps(
+    const [props, items] = useSelectionProps(
         info,
-        (s): s is (InputPort | OutputPort) => (
-            (s instanceof InputPort && s.getWires().length === 0) ||
-            s instanceof OutputPort
+        (s): s is (Component | InputPort | OutputPort) => (
+            (s instanceof Component ||
+             s instanceof InputPort && s.getWires().length === 0) ||
+             s instanceof OutputPort && s.getWires().length === 0
         ),
-        (s) => ({ type: (s instanceof InputPort ? "input" : "output") })
+        (s) => ({ type: (s instanceof Component ? "component" : "port") })
     );
 
+    // No valid selections
     if (!props)
         return null;
 
-    const iports = ps.filter(s => s instanceof InputPort)  as InputPort[];
-    const oports = ps.filter(s => s instanceof OutputPort) as OutputPort[];
+    let iports = items.filter(s => s instanceof InputPort)  as InputPort[];
+    let oports = items.filter(s => s instanceof OutputPort) as OutputPort[];
+    const components = items.filter(s => s instanceof Component) as Component[];
+
+    // Introduces ambiguity between components and individual ports
+    if ( (iports.length > 0 || oports.length > 0) && components.length > 0) {
+        return null;
+    }
+
+    // No individual ports selected, only components are selected
+    //  Bus components:
+    if (iports.length === 0 && oports.length === 0) {
+        const ambiComps = components.filter(s => s.getPorts().some(p => p instanceof InputPort &&
+                                                                        p.getWires().length === 0) ===
+                                                 s.getPorts().some(p => p instanceof OutputPort &&
+                                                                        p.getWires().length === 0)) as Component[];
+        // Remove ambiguous components that have no available ports
+        //  (occurs when an input/output only component is already connected to other components
+        //   and thus 0 free iports === 0 free oports)
+        ambiComps.forEach(c => {
+            if (c.getPorts().filter(p => p instanceof InputPort && p.getWires().length === 0).length === 0 &&
+                c.getPorts().filter(p => p instanceof OutputPort && p.getWires().length === 0).length === 0) {
```

Comment:
I don't think there should be a need to separately check that all the `InputPort`s and `OutputPort`s are full, since `.getPorts()` is returning type `Port` and the only subclasses it has in digital are `InputPort` and `OutputPort`. So I think the condition can be simplified to `c.getPorts().filter(p => p.getWires().length === 0).length === 0` and then a bit further to `c.getPorts().every(p => p.getWires().length > 0)`